qutebrowser/browser/commands.py

```diff
@@ -523,6 +523,39 @@ def tab_detach(self):
         cur_widget = self._current_widget()
         self._tabbed_browser.close_tab(cur_widget, add_undo=False)
 
+    @cmdutils.register(instance='command-dispatcher', scope='window')
+    @cmdutils.argument('index', completion=miscmodels.buffer)
+    def tab_take(self, index):
+        """Take a tab from another window.
+
+        Args:
+            index: The [win_id/]index of the tab to take. Or a substring
+                   in which case the closest match will be focused.
+        """
+        tabbed_browser, tab = self._resolve_buffer_index(index)
+
+        if tabbed_browser == self._tabbed_browser:
+            raise cmdexc.CommandError("Can't take a tab from the same window")
+
+        self._open(tab.url(), tab=True)
+        tabbed_browser.close_tab(tab, add_undo=False)
+
+    @cmdutils.register(instance='command-dispatcher', scope='window')
+    @cmdutils.argument('win_id', completion=miscmodels.window)
+    def tab_give(self, win_id: int):
+        """Give the current tab to another window.
+
+        Args:
+            win_id: The window ID of the window to give the current tab to.
+        """
+        if win_id == self._win_id:
+            raise cmdexc.CommandError("Can't give a tab to the same window")
+
+        tabbed_browser = objreg.get('tabbed-browser', scope='window',
+                                    window=win_id)
+        tabbed_browser.tabopen(self._current_url())
```

Comment:
maybe beyond the scope of this PR, but is there a way to give a tab without refreshing it? This is a (small) annoyance of mine with `tab-detach`.

Reply:
See #2636 - @Enzime originally wanted to do this as part of this PR, but it turned out to be harder than we expected. I think he still plans to look into it at some point.

Reply:
Yeah definitely planning to do it, but I'll do it for `:tab-detach`, `:tab-give` and `:tab-take` all at the same time in another PR.